From a231c8267532ee3d32ceab6ef5cf58b7b1750bd6 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:49 -0400
Subject: [usb] ehci: check the right uframes for CSPLIT

Message-id: <1410887463-20674-38-git-send-email-dzickus@redhat.com>
Patchwork-id: 95136
O-Subject: [RHEL7 PATCH 037/591] USB: EHCI: check the right uframes for CSPLIT
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit e24371a6be9c5c00b56607bd425cc409cba75d88
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Fri Oct 11 11:28:12 2013 -0400

    USB: EHCI: check the right uframes for CSPLIT

    The check_intr_schedule() routine in ehci-hcd looks at the wrong
    microframes when checking to see if a full-speed or low-speed
    interrupt endpoint will fit in the periodic schedule.  If the
    Start-Split transaction is scheduled for microframe N then the
    Complete-Split transactions get scheduled for microframes N+2, N+3, and
    N+4.  However the code considers N+1, N+2, and N+3 instead.

    This patch fixes the limits on the "for" loop and also improves the
    use of whitespace.

    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ehci-sched.c b/drivers/usb/host/ehci-sched.c
index dcbaad9..34b5945 100644
--- a/drivers/usb/host/ehci-sched.c
+++ b/drivers/usb/host/ehci-sched.c
@@ -761,7 +761,7 @@ static int check_intr_schedule (
   unsigned i;
 
   /* TODO : this may need FSTN for SSPLIT in uframe 5. */
-  for (i=uframe+1; i<8 && i<uframe+4; i++)
+  for (i = uframe+2; i < 8 && i <= uframe+4; i++)
    if (!check_period (ehci, frame, i,
       qh->period, qh->c_usecs))
     goto done;
-- 
1.7.1